File: 06 - Exam Analysis/Vcl/other/w2w/1StClass4000/VCL6/cdemos/Buttons/mapu.cpp
//---------------------------------------------------------------------------
#include <vcl.h>
#pragma hdrstop

#include "mapu.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "fcButton"
#pragma link "fcImgBtn"
#pragma link "fcLabel"
#pragma resource "*.dfm"
TMapForm *MapForm;
//---------------------------------------------------------------------------
__fastcall TMapForm::TMapForm(TComponent* Owner)
        : TForm(Owner)
{
}
//---------------------------------------------------------------------------


void __fastcall TMapForm::FormShow(TObject *Sender)
{
  //Hide the image if there is not enough colors to display the background.
  if ((GetDeviceCaps(Image2->Canvas->Handle, BITSPIXEL) *
       GetDeviceCaps(Image2->Canvas->Handle, PLANES)) <= 8)
     Image2->Visible = False;
}
//---------------------------------------------------------------------------

void __fastcall TMapForm::fcImageBtn2Click(TObject *Sender)
{
  //Set the title based on the clicked on buttons caption.
  fcLabel4->Caption = ((TfcImageBtn *)Sender)->Caption;

  //Now perform the locate on the table to get the county statistics
  TLocateOptions loptions;
  loptions.Clear();
  loptions << loPartialKey;
  Table1->Locate("CountyName",fcLabel4->Caption,loptions);
}
//---------------------------------------------------------------------------

File: 06 - Exam Analysis - New/Vcl/1StClass4000/VCL6/demos/Buttons/mapu.pas
unit mapu;

interface

uses
  Windows, Messages, SysUtils, Classes, Graphics, Controls, Forms, Dialogs,
  fcButton, fcImgBtn, ExtCtrls, fcImageForm, StdCtrls, Mask, Db, DBTables,
  fcShapeBtn, fcImage, DBCtrls, fcLabel, fcClearPanel, fcButtonGroup;

type
  TMapForm = class(TForm)
    fcImageBtn1: TfcImageBtn;
    fcImageBtn2: TfcImageBtn;
    fcImageBtn7: TfcImageBtn;
    fcImageBtn8: TfcImageBtn;
    fcImageBtn9: TfcImageBtn;
    fcImageBtn10: TfcImageBtn;
    DataSource1: TDataSource;
    Table1: TTable;
    Table1CountyName: TStringField;
    Table1SalesTaxRate: TCurrencyField;
    Table1NumberOfCities: TIntegerField;
    Table1EstimatedPopulation: TIntegerField;
    Table1EstimatedHousingUnits: TIntegerField;
    Table1CountyCode: TStringField;
    Panel1: TPanel;
    Label1: TLabel;
    DBEdit1: TDBEdit;
    Label2: TLabel;
    DBEdit2: TDBEdit;
    Label4: TLabel;
    DBEdit4: TDBEdit;
    Label6: TLabel;
    DBEdit6: TDBEdit;
    Table1CityTownList: TMemoField;
    Image2: TImage;
    fcLabel4: TfcLabel;
    fcImageBtn3: TfcImageBtn;
    fcImageBtn4: TfcImageBtn;
    fcImageBtn5: TfcImageBtn;
    fcImageBtn6: TfcImageBtn;
    DBMemo1: TDBMemo;
    Label3: TLabel;
    procedure fcShapeBtn1Click(Sender: TObject);
    procedure fcImageBtn6Click(Sender: TObject);
    procedure FormShow(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

var
  MapForm: TMapForm;

implementation

{$R *.DFM}


procedure TMapForm.fcShapeBtn1Click(Sender: TObject);
begin
  Close;
end;

procedure TMapForm.fcImageBtn6Click(Sender: TObject);
begin
  fcLabel4.Caption := (Sender as TFcImageBtn).Caption;
  case (Sender as TfcImageBtn).Tag of
    1:Table1.Locate('CountyName','Marin',[loPartialKey]);
    2:Table1.Locate('CountyName','Sonoma',[loPartialKey]);
    3:Table1.Locate('CountyName','Napa',[loPartialKey]);
    4:Table1.Locate('CountyName','Solano',[loPartialKey]);
    5:Table1.Locate('CountyName','Contra Costa',[loPartialKey]);
    6:Table1.Locate('CountyName','Alameda',[loPartialKey]);
    7:Table1.Locate('CountyName','Santa Clara',[loPartialKey]);
    8:Table1.Locate('CountyName','Santa Cruz',[loPartialKey]);
    9:Table1.Locate('CountyName','San Mateo',[loPartialKey]);
    10:Table1.Locate('CountyName','San Francisco',[loPartialKey]);
  end;
end;

procedure TMapForm.FormShow(Sender: TObject);
begin
  // Don't Show Background if 256 or less Colors
  if ((GetDeviceCaps(Image2.Canvas.Handle, BITSPIXEL) *
       GetDeviceCaps(Image2.Canvas.Handle, PLANES)) <= 8) then
     Image2.Visible := False;
end;

end.
